### PR TITLE
[DateCalendar] Remove the string argument of the `dayOfWeekFormatter` prop

### LINKED ISSUE #10991: [pickers] v7: Remove the string argument of the `dayOfWeekFormatter` prop
Part of #7902
Follow up on #9984 

**Search keywords**:

### PR DESCRIPTION
**Pull request by flaviendelangle**
Closes #10991

### CODE REVIEW

File: docs/data/migration/migration-pickers-v6/migration-pickers-v6.md [hunk 1]
```diff
@@ -63,3 +63,21 @@ For example:
...
+
+```diff
+ <DateCalendar
+   // If you were still using the day string
+-   dayOfWeekFormatter={dayStr => `${dayStr}.`}
++   dayOfWeekFormatter={day => `${day.format('dd')}.`}
+
+   // If you were already using the day object
+-   dayOfWeekFormatter={(_dayStr, day) => `${day.format('dd')}.`
++   dayOfWeekFormatter={(day) => `${day.format('dd')}.`
```
Comment (michelengelen) [inline docs/data/migration/migration-pickers-v6/migration-pickers-v6.md:81]: Only a nit
```suggestion
+   dayOfWeekFormatter={day => `${day.format('dd')}.`
```

**Review by michelengelen — APPROVED** [review]
LGTM

File: docs/data/migration/migration-pickers-v6/migration-pickers-v6.md [hunk 2]
```diff
@@ -82,6 +82,24 @@ The imports related to the `calendarHeader` slot have been moved from `@mui/x-da
...
 
+## Modified props
+
+### Remove the string argument of the `dayOfWeekFormatter` prop
+
+The string argument of the `dayOfWeekFormatter` has been replaced in favor of the date object to allow more flexibility.
+
+```diff
+ <DateCalendar
+   // If you were still using the day string
```
Comment (alexfauquette) [inline docs/data/migration/migration-pickers-v6/migration-pickers-v6.md:93]:
```suggestion
   // If you were still using the day string, you can get it back with your date library.
```

File: docs/data/migration/migration-pickers-v6/migration-pickers-v6.md [hunk 3]
```diff
@@ -82,6 +82,24 @@ The imports related to the `calendarHeader` slot have been moved from `@mui/x-da
...
+-   dayOfWeekFormatter={dayStr => `${dayStr}.`}
++   dayOfWeekFormatter={day => `${day.format('dd')}.`}
+
+   // If you were already using the day object
```
Comment (alexfauquette) [inline docs/data/migration/migration-pickers-v6/migration-pickers-v6.md:97]:
```suggestion
   // If you were already using the day object, just remove the first argument.
```

**Review by alexfauquette — APPROVED** [review]
Don't think it is worth a codemod.

I proposed picky modification of the breaking change. Not sure either they are an improvement or not

**Pull request merged**